server/src/test/java/org/opensearch/indices/replication/SegmentReplicationTargetServiceTests.java

```diff
@@ -77,44 +78,58 @@ public void setUp() throws Exception {
 
     @Override
     public void tearDown() throws Exception {
-        closeShards(indexShard);
+        closeShards(primaryShard, replicaShard);
         super.tearDown();
     }
 
-    public void testTargetReturnsSuccess_listenerCompletes() {
-        final SegmentReplicationTarget target = new SegmentReplicationTarget(
-            checkpoint,
-            indexShard,
-            replicationSource,
-            new SegmentReplicationTargetService.SegmentReplicationListener() {
-                @Override
-                public void onReplicationDone(SegmentReplicationState state) {
-                    assertEquals(SegmentReplicationState.Stage.DONE, state.getStage());
-                }
+    public void testsSuccessfulReplication_listenerCompletes() throws InterruptedException {
+        CountDownLatch latch = new CountDownLatch(1);
+        sut.startReplication(checkpoint, replicaShard, new SegmentReplicationTargetService.SegmentReplicationListener() {
+            @Override
+            public void onReplicationDone(SegmentReplicationState state) {
+                assertEquals(SegmentReplicationState.Stage.DONE, state.getStage());
+                latch.countDown();
+            }
 
-                @Override
-                public void onReplicationFailure(SegmentReplicationState state, OpenSearchException e, boolean sendShardFailure) {
-                    Assert.fail();
-                }
+            @Override
+            public void onReplicationFailure(SegmentReplicationState state, OpenSearchException e, boolean sendShardFailure) {
+                logger.error("Unexpected error", e);
+                Assert.fail("Test should succeed");
             }
-        );
-        final SegmentReplicationTarget spy = Mockito.spy(target);
-        doAnswer(invocation -> {
-            // set up stage correctly so the transition in markAsDone succeeds on listener completion
-            moveTargetToFinalStage(target);
-            final ActionListener<Void> listener = invocation.getArgument(0);
-            listener.onResponse(null);
-            return null;
-        }).when(spy).startReplication(any());
-        sut.startReplication(spy);
+        });
+        latch.await(2, TimeUnit.SECONDS);
```

Comment:
nit: 2 secs wait seems too long, does 100ms is sufficient, as this is unit test. Applicable for other UTs as well.

Reply:
Given that this only waits 2s in the failure case & these rely on some async steps, I'd prefer to keep this higher so we aren't introducing flakey tests.